<a id="des3"> </a>
# Desafío 3: Convertidor de temperaturas
Realizar conversiones de una temperatura dada en Celsius a Fahrenheit y Kelvin utilizando operaciones directas sobre las variables. Los cálculos se realizarán directamente al asignar los valores a las variables.

Explicación del ejercicio: Celda 1: Elegimos una temperatura, por ejemplo 25ºC. Celda 2: Convertir a Fahrenheit. Celda 3: Converitr a Kelvin. Celda 4: Mostrar los resultados.
Fórmulas que vamos a utulizar: 
- Fahrenheit = (Celcius x 9/5) + 32
- Kelvin = Celcius + 273.15

In [1]:
# Definimos la temperatura en Celcius
celsius = 25
celsius

# Convertimos de Celcius a Fahrenheit
fahrenheit = (celsius * 9/5) + 32
fahrenheit

# Convertimos de Celsius a Kelvin
kelvin = celsius + 273.15
kelvin

# Mostramos todas las conversiones
print(f"Temperatura en Celsius: {celsius}ºc")
print(f"Equivalente en Fahrenheit: {fahrenheit: .2f}ºF")
print(f"Equivalente en Kelvin: {kelvin: .2f}K")

Temperatura en Celsius: 25ºc
Equivalente en Fahrenheit:  77.00ºF
Equivalente en Kelvin:  298.15K


Tipos de datos usados:
float → para representar temperaturas (ej. 25.0, 298.15).

Operadores usados:
* → multiplicación.
/ → división.
+ → suma.
f-string → para mostrar texto con variables dentro (f"...").

<a id="des4"> </a>
# Desafío 4: Verificar múltiplos de varios números
Objetivo: Determinar si un número dado es múltiplo de 2, 3, 5, 7, 9, 10, y 11 utilizando comparaciones y operadores lógicos.

*Descripción:*
Crea un programa que, dado un número, verifique si es múltiplo de los números mencionados. Para cada uno de estos números, el programa debe indicar si el número dado es o no un múltiplo de estos.

_Ejemplo de lo que se debe lograr:_
Supongamos que el número dado es 30. El programa deberá determinar:

¿Es 30 múltiplo de 2? Sí

¿Es 30 múltiplo de 3? Sí

¿Es 30 múltiplo de 5? Sí

¿Es 30 múltiplo de 7? No

¿Es 30 múltiplo de 9? No

¿Es 30 múltiplo de 10? Sí

¿Es 30 múltiplo de 11? No

Explicación paso a paso del ejercicio: Verificar si un número es múltimplo de otros, en mi ejemplo voy a utulizar el número 20.
Paso 1: Definir el número, esto simplemente guarda el numero en la variable (numero), que luego vamos a usar para hacer las comprobaciones. Paso 2: numero % 2 == 0, % es el operador módulo, que da el resto de una division, si el resto es 0, eso significa que el número es múltiplo. entonces 20%2 da 0: es múltiplo de 2, si hicieramos 20%3 que da 2, entonces no es múltiplo de 3.
Paso 3: 'Sí' if numero % 2 == 0 else 'No', esto es una forma compacta de escribir if; Si numero % 2 == 0 es verdadero devuelve 'Si', sino, devuelve 'No'.
Paso 4: Imprimir el resultado en una frase completa: print(f"¿Es {numero} múltiplo de 2? {'Sí if numero % 2 == 0 else 'No'}"), ¿Qué hace esto? usa una f-string (f"") para combinar texto con variables y cálculos. El resultado se vería así: ¿Es 20 múltiplo de 2? Sí. se repite el procedimiento con los otros números. 
Paso 5: Resultado final con numero = 20: un ejemplo sería ¿Es 20 múltiplo de 3? No, y asi sucesivamente hasta terminar la lista de números que tengamos.

In [2]:
# Números a verificar
2, 3 , 5, 7, 9, 10, 11

# Definimos el número que vamos a verificar
numero = 20

# Verificamos si es múltiplo de los distintos valores

print(f"¿Es {numero} múltiplo de 2? {'Sí' if numero % 2 == 0 else 'No'}")
print(f"¿Es {numero} múltiplo de 3? {'Sí' if numero % 3 == 0 else 'No'}")
print(f"¿Es {numero} múltiplo de 5? {'Sí' if numero % 5 == 0 else 'No'}")
print(f"¿Es {numero} múltiplo de 7? {'Sí' if numero % 7 == 0 else 'No'}")
print(f"¿Es {numero} múltiplo de 9? {'Sí' if numero % 9 == 0 else 'No'}")
print(f"¿Es {numero} múltiplo de 10? {'Sí'if numero % 10 == 0 else 'No'}")
print(f"¿Es {numero} múltiplo de 11? {'Sí'if numero % 11 == 0 else 'No'}")

¿Es 20 múltiplo de 2? Sí
¿Es 20 múltiplo de 3? No
¿Es 20 múltiplo de 5? Sí
¿Es 20 múltiplo de 7? No
¿Es 20 múltiplo de 9? No
¿Es 20 múltiplo de 10? Sí
¿Es 20 múltiplo de 11? No


¿Qué tipo de datos y operadores utilizaste?

Tipos de datos usados:
int → el número que se analiza (ej. 20).
str → para imprimir “Sí” o “No”

Operadores usados:
% (módulo) → para obtener el resto de una división.
== → comparación de igualdad.
if ... else (condicional inline).
f-string → para imprimir resultados con texto y variables.

No adjunto Webgrafía porque utilice el material proporcionado por el docente, es decir el material que se encuentra en los cuadernos: 1_2, 1_3 y sobre todo el 2_4.